PNA.mdk

```diff
@@ -930,7 +930,9 @@ property is `true` for a table, the P4 developer is allowed to define
 a default action for the table that calls the `add_entry` extern
 function.  `add_entry` adds a new entry to the table whose default
 action calls the `add_entry` function.  The new entry will have the
-same key that was just looked up.
+same key that was just looked up. The new entry value, as in a (key,value)
+pair, is obtained from network headers of the packet that caused the
+table lookup to miss.
```

Comment:
It _might_ be obtained from network headers of the packet, but it is actually obtained from the key that was looked up, as defined in the key definition of the table.  It is allowed to include _any_ fields, whether they be metadata, packet header fields, or any other expressions allowed as table key fields in a P4 table.  The way you have written it here, it is overly restrictive.

Reply:
What do you want to say that "The new entry will have the same key that was just looked up." does not say?

Reply:
Isn't metadata refreshed for every packet? This why I skipped mentioning metadata. Maybe not.  The table key is just a key in (key, value) store, and this key lookup failed, so no key, value store exists to copy its value data for new entry. I like "other expressions" text you proposed.

Reply:
When I wrote "the key that was looked up" in my original text, I meant the value that is called `lookupKey` in this section of the language spec: https://p4.org/p4-spec/docs/P4-16-v-1.2.3.html#sec-mau-semantics

That value might match in the existing table entries, or it might miss.  If it misses, that `lookupKey` value is the new one to be added to the table by the call to `add_entry`.

If any of that is clearer or more precise than what I wrote, I'm open to a PR that suggests words including the above.

Reply:
Even if user-defined metadata is refreshed for every packet, your P4 program can make assignments to metadata, e.g. based upon table lookups, _before_ reaching the table with add_on_miss=true, so it can be very useful to include in the key of such a table (e.g. a VRF id derived from a table lookup, not in the packet).

Also, some intrinsic metadata like the input port/vport of the packet might be useful in the key of an add_in_miss=true table.